module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp

```diff
@@ -577,18 +571,13 @@ namespace bsp::battery_charger
         // Short time to synchronize after configuration
         vTaskDelay(pdMS_TO_TICKS(100));
         StateOfCharge level = getBatteryLevel();
-        bool charging       = getChargeStatus();
-        LOG_INFO("Phone battery start state: %d %d", level, charging);
 
         clearAllChargerIRQs();
         clearFuelGuageIRQ(static_cast<std::uint16_t>(batteryINTBSource::all));
         enableTopIRQs();
         enableChargerIRQs();
         IRQPinsInit();
 
-        // Check IRQ status register
-        INTB_IRQHandler();
-
```

Comment:
That was crucial in the init function. Was it moved somewhere or just deleted?

Reply:
The initial state of SOC, battery voltage, and charger is fetched in the higher layers(BatteryController) so there is no longer a need to check it here.

Reply:
Main reason was to clear status register -> can you check if it's happening anywhere else?

Reply:
Taken from MAX77818 init code:
```
        clearAllChargerIRQs();
        clearFuelGuageIRQ(static_cast<std::uint16_t>(batteryINTBSource::all));
        enableTopIRQs();
        enableChargerIRQs();
        IRQPinsInit();
```
it looks alright to me. All IRQ status registers are cleared before enabling specific irqs. Then it comes board-specific IRQ pins init. At this point, it is safe to send IRQ notifications to the WorkerQueue as it was initialized before the MAX77818 driver. Ideally, it would be better to make MAX77818 driver board-agnostic just like the CW2015 and MP2639B.